### ISSUE TITLE
Missing file InstallController.php in branch dev

### ISSUE THREAD

**Issue by guerriky** [userfrosting/UserFrosting#548]
Hello,
i saw you introduced a bunch of changes to the system in branch dev.

I'd have liked to try them out, but unfortunately the install.php route (app/routest/install.php line 27) outputs a fatal error for a missing InstallController file.

Could you provide said file, or a path on which I can check it out?

Thank you,
Riccardo

**Comment by Netrilix** [comment]
It's not that the file is missing, it's that you need to re-run `composer update` so the autoloader will know where to find the new classes.

**Comment by frostbitten** [comment]
Actually, I believe there is no such controller. You would need to modify those lines here:

https://github.com/frostbitten/UserFrosting/blob/7c126bdd183b661d20ec16e5eeb8bfefbb2dcb59/app/routes/install.php#L26-L27

**Comment by alexweissman** [comment]
@guerriky the current `dev` branch is abandoned at the moment.  We're rebuilding UF from the ground up at https://github.com/userfrosting/bones, if you want to check that out.

**Comment by guerriky** [comment]
Hello everyone,
thank you for your answers.

@frostbitten I tried that and it works fine, until another error catches up (I suspect another pending dependency) but nevermind, I'll keep my eye on the new bones repository.

@alexweissman I cloned it and it works fine, i see the last commit is, like, three days ago, so it is under development.

For now, I'll just extract the features I'm interested in from this dev branch.

Thanks again,
Riccardo

**Comment by alexweissman** [comment]
Cool, sounds good.  Just out of curiosity, which features are you interested in, in particular?

**Comment by guerriky** [comment]
Hi Alex,
my idea was to make some changes to the original source in a way that would have added the possibility to add pages from the database, kinda like how Mediawiki works.

That'd make a lot of my project easier, since I'll most likely need some sort of page creation GUI and, in the future, a page creation backend.

One of the targets that I set up for to do this was to trim down the index.php in favor of a more flexible "view switch" (however that's called) and I saw just that in the dev branch.
That's a big plus, since there's no more need to add a bunch of code in the index direct, just to make it work.

Another thing I'd have liked would be extending the Fortress validation to accept wildcards in the field name. 
The real purpose of this is to free the individual pages from the hassle of creating the json files themselves.
While not a big difference, it means a lot to my systems.

Other things would be generic views, frontend field creation, table CSV export and so on, but I don't think there's anything done yet.

When I'm done, I might be able to push something to the new repo. If I don't screw the whole thing in the attempt :) .

Thank you,
Riccardo

**Comment by alexweissman** [comment]
@guerriky what kind of pages do you need to create?  Static content?  Have you looked into any CMS systems available on Github?

**Comment by guerriky** [comment]
Hi @alexweissman ,
the idea is this: we have some SBCs with limited capabilities.

On these systems, 99% of the database work means either logging to a table or updating a list of fields (sensor data and the like).
When the time to add new resources come, this is generally done by adding a table, for hardware reasons.

What I need to do is to provide two "generic" views that would allow for any user to connect to a url like "userfrosting/public/pages/foobar" and be able to see the table content without writing a single line of code.

Before switching to Userfrosting, we had CakePHP, and it was great for many things, but not so much for the resources it needed (embedded systems just couldn't hande them).

We tried many others as well, but we settled on this because it's the one that matches most of our needs; the features we don't have, we try to make them :)

Thank you,
Riccardo
